integrations/pagerduty/webhook_processors/incidents.py

```diff
@@ -0,0 +1,39 @@
+from clients.pagerduty import PagerDutyClient
+from kinds import Kinds
+from webhook_processors.abstract import (
+    PagerdutyAbstractWebhookProcessor,
+)
+from port_ocean.core.handlers.port_app_config.models import ResourceConfig
+from port_ocean.core.handlers.webhook.webhook_event import (
+    EventPayload,
+    WebhookEvent,
+    WebhookEventRawResults,
+)
+
+
+class IncidentWebhookProcessor(PagerdutyAbstractWebhookProcessor):
+    async def should_process_event(self, event: WebhookEvent) -> bool:
+        client = PagerDutyClient.from_ocean_configuration()
+        return (
+            event.payload.get("event", {}).get("event_type")
+            in client.incident_upsert_events
+        )
+
+    async def get_matching_kinds(self, event: WebhookEvent) -> list[str]:
+        return [Kinds.INCIDENTS]
+
+    async def handle_event(
+        self, payload: EventPayload, resource_config: ResourceConfig
+    ) -> WebhookEventRawResults:
+        client = PagerDutyClient.from_ocean_configuration()
+        incident_id = payload.get("event", {}).get("data", {}).get("id")
+        incident = await client.get_singular_from_pager_duty(
```

Comment:
i know you didn't write it, but I am not a fan of the `get_singular_from_pager_duty` lets use similar terminology to jira / github with `get_single_resource`